python_modules/dagster/dagster/core/asset_defs/assets.py

```diff
@@ -92,3 +134,55 @@ def get_partition_mapping(self, in_asset_key: AssetKey) -> PartitionMapping:
             in_asset_key,
             self._partitions_def.get_default_partition_mapping(),
         )
+
+
+def _infer_asset_keys_by_input_names(
+    graph_def: GraphDefinition, asset_keys_by_input_name: Mapping[str, AssetKey]
+) -> Mapping[str, AssetKey]:
+    all_input_names = {graph_input.definition.name for graph_input in graph_def.input_mappings}
+    for in_key in asset_keys_by_input_name.keys():
+        if in_key not in all_input_names:
+            raise DagsterInvalidDefinitionError(
+                f"Key '{in_key}' in provided asset_keys_by_input_name dict does not correspond to "
+                "any key provided in the ins dictionary of the decorated op or any argument "
+                "to the decorated function"
+            )
+
+    # If asset key is not supplied in asset_keys_by_input_name, create asset key
+    # from input name
+    inferred_input_names_by_asset_key: Dict[str, AssetKey] = {
+        input_name: asset_keys_by_input_name.get(input_name, AssetKey([input_name]))
+        for input_name in all_input_names
+    }
+
+    return inferred_input_names_by_asset_key
+
+
+def _infer_asset_keys_by_output_names(
+    graph_def: GraphDefinition, asset_keys_by_output_name: Mapping[str, AssetKey]
+) -> Mapping[str, AssetKey]:
+    inferred_asset_keys_by_output_names: Dict[str, AssetKey] = {
+        output_name: asset_key for output_name, asset_key in asset_keys_by_output_name.items()
+    }
+    output_names = [output.definition.name for output in graph_def.output_mappings]
```

Comment:
why graph_def.output_mappings here instead of just graph_def.output_defs?

it might work the same but I'd be at least slightly concerned that weird edge cases could arise where the op's output name differs from the graph's output name (we want to use the graph's output name)

Reply:
No particular reason here--just saw the `output_mappings` arg on graph definition. But i've adjusted this to use the `output_defs`